src/main/java/com/example/controller/FileMenuController.java

```diff
@@ -18,12 +22,24 @@
 public class FileMenuController {
     @Autowired
     GraphController graphController;
+    @Autowired
+    AppController appController;
+    @FXML
+    MenuItem newFileButton;
     @FXML
     MenuItem importButton;
     @FXML
     MenuItem exportButton;
 
     private void bindButtons() {
+        ObjectProperty<ApplicationState> applicationState = appController.getApplicationStateProperty();
+        BooleanBinding isSimulation = applicationState.isEqualTo(ApplicationState.SIMULATING);
+
+        newFileButton.disableProperty().bind(isSimulation);
+
+        newFileButton.setOnAction(e -> {
+            graphController.setModelGraph(new MyGraph<>());
```

Comment:
I've noticed one more problem. We update our graph in graphController, however we have reference to an old graph in appController. Please check if we could remove graph reference from appController (I mean don't hold such reference). AppController culd probably only instantiate new graph in initGraph() method.

Reply:
Removed.